### ISSUE TITLE
Inquiry on Rails 7.2 Support Effort

### ISSUE THREAD

**Issue by seuros** [rgeo/activerecord-postgis-adapter#395]
Hi everyone,

I wanted to check if anyone is currently working on adding support for Rails 7.2. If there is already an effort underway, I would love to know more about it and see if there's any way I can assist.

If no one is handling this task, I am planning to dedicate some time over the weekend to work on it. 

Remember this will be again a breaking change since the new version will support just 7.2.x

**Comment by theonlyriddle** [comment]
🥳 This would be very helpful! 

I just updated to Rails 7.2 and found that my application wouldn't boot up. We've tentatively narrowed down the culprit to this gem.

**Comment by anthony0030** [comment]
@seuros that would be sooo Cool, I was debugging an app all day yesterday and came to the conclusion that just changing the version number in this gem was not enough to get it working.

**Comment by ugisozols** [comment]
👋 I was also looking into this and got my Rails app booting with the changes in https://github.com/rgeo/activerecord-postgis-adapter/pull/396. I do not know if that's enough at this point as my Rails app needs many more updates to run on 7.2, so take it as is.

**Comment by keithdoggett** [comment]
Hey everyone thanks for the help in getting this up! Unfortunately I will not have much time to contribute for the next few weeks but will have time to review/make a release once we have a working version.

@ugisozols thanks for starting the PR. I haven't reviewed it yet, but we can work off of that branch for now and if necessary I can make a different dev branch people can fork off of.

**Comment by BuonOmo** [comment]
@seuros it looks like the fix of require `rgeo/activerecord` was not enough (https://github.com/rgeo/activerecord-postgis-adapter/actions/runs/9805704288/job/27075934099). Do you think you would have time to fix that ? I think We'll have to wait for this before we can ship unfortunately :/

**Comment by seuros** [comment]
@BuonOmo  That a flaky test happening only in CI and don't affect runtime (the connection take some time to get established), restart the specific build it will pass.

As i said in the other discussion, i will fix it in a separate PR.

**Comment by StoneGod** [comment]
Hi! What about releasing new version?

**Comment by StoneGod** [comment]
@seuros Hi! what about release new version of gem? https://github.com/rgeo/activerecord-postgis-adapter/pull/403
rails 7.2 already released

**Comment by totus** [comment]
Same Q here

**Comment by seuros** [comment]
The code is already merged in this repo, if you could use github path to run you whole test suite that will be helpful.

We don't want to release something that is not battle tested.

The test suite is flaky, but works locally.

**Comment by JamesChevalier** [comment]
> The code is already merged in this repo, if you could use github path to run you whole test suite that will be helpful.
> 
> We don't want to release something that is not battle tested.
> 
> The test suite is flaky, but works locally.

I bumped my Rails version in my Gemfile:

```
gem "rails", "~> 7.2.1"
```

And pointed activerecord-postgis-adapter to GitHub:

```
gem "activerecord-postgis-adapter", github: "rgeo/activerecord-postgis-adapter"
```

And my test suite produces the error reported in https://github.com/rgeo/activerecord-postgis-adapter/issues/402 :

```
TypeError:
  can't cast RGeo::Geographic::SphericalLineStringImpl
```

I have some details that might be helpful that I'll share in that Issue.

**Comment by mjy** [comment]
For what it's worth we have a 7.2 unofficial PR branch in CI for our app, see https://github.com/SpeciesFileGroup/taxonworks/pull/4006.  

There are quite a few geo specs in the suite  (though I wouldn't recommend them as a model for how to do things, we have another branch refactoring the whole approach), and the failures we do have seem unrelated.

**Comment by robisonsantos** [comment]
When should we have a new release of the gem?

**Comment by mjy** [comment]
@robisonsantos @StoneGod "When" questions on open source efforts of this nature are often frowned upon, this is all volunteer work for which we are very thankful. To express that you'd appreciate this work being done, it's common to add a "thumbs-up" to the initial comment on an issue like this rather than tacking on a comment.

**Comment by robisonsantos** [comment]
It wasn't my intention to offend anyone and I apologize if I did. It was
just a question based on the fact the fix seems in place but the official
gem does not yet track the latest changes.

**Comment by StoneGod** [comment]
> @robisonsantos @StoneGod "When" questions on open source efforts of this nature are often frowned upon, this is all volunteer work for which we are very thankful. To express that you'd appreciate this work being done, it's common to add a "thumbs-up" to the initial comment on an issue like this rather than tacking on a comment.

If anyone here is offended by the word “when,” then I apologize.

I opened MP with the update to 7.2, but later I saw that work was already underway in this direction. And the problem lies in the tests, I would be happy to help, but I have not yet fully immersed myself in the structure of the tests.

**Comment by BuonOmo** [comment]
@mjy thank you for clarifying. Indeed the gem is only volunteerely maintained. I once was paid by the company I worked at to do so, it is not the case anymore. If this gem is important to your company, you can find my email on my username, and a sponsorship link on the page project. I'm willing to work for free, and I'll keep maintaining the gem. But if you expect deadlines (e.g. "when"), ask your boss to pay the price!

I've seen some people started helping on with the tests. I have some paid work to do today, I may have some time during the week to review. I'll have time to answer every issue or PR. Maybe Keith will have some time as well, but his company is also not paying him for this maintaining job. I think I'll add a post-install message to this and the rgeo gem looking for support, we're used by loads of governements and big companies, it is time to pay for the service!

In the meantime, feel free to use the branch in developement, and report issues if they come up.

**Comment by anthony0030** [comment]
First of all, I would like to thank all the maintainers of this project and volunteers. I had the same question about when this Issue would be closed, but now that I am aware that it is not good to ask this question, I won't.

@BuonOmo your latest reply has raised some questions for me and possibly others reading:
`But if you expect deadlines (e.g. "when"), ask your boss to pay the price!`
Could you let me know the price to get 7.2 supported as soon as possible?
What funding/donations are needed to get the soon-to-be-announced V8 compatible?

Thank you again for your hard work. We appreciate you a lot.

**Comment by seuros** [comment]
@anthony0030 7.2 is supported, depending on your use case.

You can use the GitHub reference to install it and check if your tests pass.  

But we won't release a final version until the flaky bugs are fixed.

Releasing a final version means that we green-lighted the upgrade for other devs, without any action on their behalf (or documented upgrade instructions)


As for Version 8, it technically possible to run it by changing the gemspec constraints. 

But like with 7.1 -> 7.2 , the ActiveRecord api changed in the last months and it broke compatibility. So use it at your own risk.

**Comment by BuonOmo** [comment]
> Could you let me know the price to get 7.2 supported as soon as possible?
> What funding/donations are needed to get the soon-to-be-announced V8 compatible?

@anthony0030 I think that paying for 7.2 support is a bit late now as the time that we'll get a contract we'd already have it done. For 8, we could discuss a deadline. As I'm not sure yet about how to finance it and I wouldn't want to clutter the conversation here, I suggest talking via emails (see my profile), I'll also add @keithdoggett to the loop.

About my personal opinion as of now (I want to grab more context on our users to be sure of our needs). Until we have a precise idea for how donation could work, if you company relies on postgis or rgeo features, I suggest a monthly donation on https://opencollective.com/rgeo. I think that 3000$ a month would be the ultimate goal if we want to have a safe and healthy ecosystem for the RGeo gems. In the meantime, the amount you can donate really depend on your company. I would thinking something like at most 10$/dev/company should be enough if a few percentage of the companies using our gems would do it. Both these numbers are to be taken with a grain of salt, we'd have to think them more thoroughly.

> But we won't release a final version until the flaky bugs are fixed.

It looks like @formigarafa made a great job of fixing them. @seuros would you be able to be the second reviewer as I'm not sure how available is @keithdoggett these days?

> But like with 7.1 -> 7.2 , the ActiveRecord api changed in the last months and it broke compatibility. So use it at your own risk.

Fully agree.  If you need some warranty about this in terms of deadline and safety, we could exchange further by email. I think that if I focus mainly on this next week it will be done at most next Friday.
